src/codegen_expr_literal.cpp

```
@@ -624,24 +714,285 @@ llvm::Value *CodeGen::valueToString(llvm::Value *val) {
+        }
 
         std::string name = structTy->getName().str();
         if (struct_types_.count(name))
             return structToString(val);
         if (isTupleStructType(structTy))
             return tupleToString(val, structTy);
+        codegenError("cannot convert this struct type to string: " + name);
+    }
+
+    // Fixed-length array: sprint buffer + IR loop
+    if (auto *ai = llvm::dyn_cast<llvm::AllocaInst>(val)) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)